plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -5314,6 +5314,24 @@ public void testInsertingIntoTablesWithColumnsWithQuotesInName()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testInsertIntoBucketedColumnWhenTaskWriterCountIsGreaterThanOrEqualToNodeCount()
+    {
+        int taskWriterCount = 4;
+        assertThat(taskWriterCount).isGreaterThanOrEqualTo(getQueryRunner().getNodeCount());
+        Session session = Session.builder(getSession())
+                .setSystemProperty("task_writer_count", String.valueOf(taskWriterCount))
+                .build();
+
+        String tableName = "test_inserting_into_bucketed_column_when_task_writer_count_is_greater_than_or_equal_to_node_count_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (bucketed_col INT) WITH (partitioning = ARRAY['bucket(bucketed_col, 10)'])");
+
+        assertUpdate(session, "INSERT INTO " + tableName + " VALUES (1)", 1);
```

Comment:
`INSERT nationkey SELECT FROM tpch.tiny.nation`
otherwise the planner could realize it's inserting exactly one row, and could limit writer count to 1 without talking to the connector.

Reply:
Also, worth adding cases with CTAS, UPDATE, DELETE and MERGE